src/agent/onefuzz-agent/src/worker.rs

```diff
@@ -130,10 +133,36 @@ impl<C: Context> State<C> {
 
 impl State<Ready> {
     pub async fn run(self, runner: &mut dyn IWorkerRunner) -> Result<State<Running>> {
-        let child = runner.run(&self.ctx.setup_dir, &self.work).await?;
+        // Create and pass the server here
+        let (from_agent_to_task_server, from_agent_to_task_endpoint) = IpcOneShotServer::new()?;
+        let (from_task_to_agent_server, from_task_to_agent_endpoint) = IpcOneShotServer::new()?;
+        let child = runner
+            .run(
+                &self.ctx.setup_dir,
+                &self.work,
+                from_agent_to_task_endpoint,
+                from_task_to_agent_endpoint,
+            )
+            .await?;
+
+        // Accept is a blocking call:
```

Comment:
If it's blocking we should use `spawn_blocking`: https://docs.rs/tokio/latest/tokio/task/fn.spawn_blocking.html